build/tasks/ProjectModel/ProjectInfoFactory.cs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using NuGet.Frameworks;
+using RepoTasks.Utilities;
+using Microsoft.Build.Utilities;
+
+namespace RepoTasks.ProjectModel
+{
+    internal class ProjectInfoFactory
```

Comment:
Can't we just shell out to `GenerateRestoreGraphFile` and adapt it to our needs? We can pretty much infer most of these from the spec file, no?

Reply:
The restore graph file was okay, but doesn't give us the complete story. Running a design-time build instead gives us the same information, and allows us to get even more, such as specific MSBuild properties like 'PackageType' and the location of obj/ and bin/.

Reply:
What do we need ouput paths for? Presumably all we need for lineups is package version information.

Reply:
Not used yet. This code was copied from the dev branch of KoreBuild from the lineups task. I'm trying to keep it as similar as possible to reduce our pain when porting this work to dev.

Reply:
Fair enough

Reply:
Also, it turns out that calculating the build graph this way is faster. It's about twice as fast as the previous method which generated a JSON file using /t:GenerateRestoreGraphFile